pkg/cloudprovider/providers/gce/gce_disks_test.go

```diff
@@ -0,0 +1,401 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gce
+
+import (
+	"testing"
+
+	"golang.org/x/net/context"
+	compute "google.golang.org/api/compute/v1"
+	"google.golang.org/api/googleapi"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/pkg/cloudprovider"
+)
+
+func TestCreateDisk_Basic(t *testing.T) {
+	fakeManager := newFakeManager()
+	gce := GCECloud{manager: fakeManager, managedZones: []string{"zone1"}}
+
+	diskName := "disk"
+	diskType := DiskTypeSSD
+	zone := "zone1"
+	const sizeGb int64 = 128
+	err := gce.CreateDisk(diskName, diskType, zone, sizeGb, nil)
+	if err != nil {
+		t.Error(err)
+	}
+	if !fakeManager.createDiskCalled {
+		t.Error("Never called GCE disk create.")
+	}
+	if !fakeManager.doesOpMatch {
+		t.Error("Ops used in WaitForZoneOp does not match what's returned by CreateDisk.")
+	}
+	// Partial check of equality between disk description sent to GCE and parameters of method.
+	if fakeManager.diskToCreate.Name != diskName || fakeManager.diskToCreate.SizeGb != sizeGb {
```

Comment:
I would also verify the tags, and diskTypeUri fields.  And make each field a separate check.